daemon/daemon.go

```diff
@@ -1062,7 +1068,7 @@ func (daemon *Daemon) PullOnBuild(name string, authConfigs map[string]types.Auth
 		pullRegistryAuth = &resolvedConfig
 	}
 
-	if err := daemon.PullImage(ref, nil, pullRegistryAuth, output); err != nil {
+	if err := daemon.PullImage(nil, ref, nil, pullRegistryAuth, output); err != nil { //FIXME:  don't pass nil here for ctx
```

Comment:
Suggest changing the nil ctx argument to `context.TODO()`. Then you can remove the check in `PullImage`.